TransformationSystem/DB/TransformationDB.sql

```diff
@@ -113,45 +113,57 @@ CREATE TABLE TransformationFiles(
-  PRIMARY KEY(TransformationID,FileID,TaskID),
-  FOREIGN KEY (TransformationID) REFERENCES Transformations(TransformationID),
-  FOREIGN KEY (TransformationID, FileID) REFERENCES TransformationFiles(TransformationID, FileID),
-  FOREIGN KEY (TransformationID, TaskID) REFERENCES TransformationTasks(TransformationID, TaskID)
-) ENGINE=InnoDB DEFAULT CHARSET=utf8;
+DROP TABLE IF EXISTS TransformationFileTasks
+CREATE TABLE TransformationFileTasks(
+    TransformationID INTEGER NOT NULL,
+    FileID INTEGER NOT NULL,
+    TaskID INTEGER NOT NULL,
+    PRIMARY KEY(TransformationID, FileID, TaskID),
+    FOREIGN KEY(TransformationID) REFERENCES Transformations(TransformationID),
+    FOREIGN KEY(TransformationID, FileID) REFERENCES TransformationFiles(TransformationID, FileID),
+    FOREIGN KEY(TransformationID, TaskID) REFERENCES TransformationTasks(TransformationID, TaskID)
+) ENGINE = InnoDB DEFAULT CHARSET = utf8
 
 -- -------------------------------------------------------------------------------
-DROP TABLE IF EXISTS TaskInputs;
-CREATE TABLE TaskInputs (
-  TransformationID INTEGER NOT NULL,
-  TaskID INTEGER NOT NULL,
-  InputVector MEDIUMTEXT,
-  PRIMARY KEY(TransformationID,TaskID),
-  FOREIGN KEY (TransformationID, TaskID) REFERENCES TransformationTasks(TransformationID, TaskID)
-) ENGINE=InnoDB DEFAULT CHARSET=utf8;
+DROP TABLE IF EXISTS TaskInputs
+CREATE TABLE TaskInputs(
+    TransformationID INTEGER NOT NULL,
+    TaskID INTEGER NOT NULL,
+    InputVector MEDIUMTEXT,
+    PRIMARY KEY(TransformationID, TaskID),
+    FOREIGN KEY(TransformationID, TaskID) REFERENCES TransformationTasks(TransformationID, TaskID)
+) ENGINE = InnoDB DEFAULT CHARSET = utf8
 
 -- -------------------------------------------------------------------------------
-DROP TABLE IF EXISTS TransformationInputDataQuery;
-CREATE TABLE TransformationInputDataQuery (
-  TransformationID INTEGER NOT NULL,
-  ParameterName VARCHAR(255) NOT NULL,
-  ParameterValue BLOB NOT NULL,
-  ParameterType VARCHAR(8) NOT NULL,
-  PRIMARY KEY (TransformationID,ParameterName),
-  FOREIGN KEY (TransformationID) REFERENCES Transformations(TransformationID)
-) ENGINE=InnoDB DEFAULT CHARSET=utf8;
+DROP TABLE IF EXISTS TransformationInputDataQuery
+CREATE TABLE TransformationInputDataQuery(
+    TransformationID INTEGER NOT NULL,
+    ParameterName VARCHAR(255) NOT NULL,
+    ParameterValue BLOB NOT NULL,
+    ParameterType VARCHAR(8) NOT NULL,
+    PRIMARY KEY(TransformationID, ParameterName),
+    FOREIGN KEY(TransformationID) REFERENCES Transformations(TransformationID)
+) ENGINE = InnoDB DEFAULT CHARSET = utf8
+
+-- -------------------------------------------------------------------------------
+DROP TABLE IF EXISTS TransformationMetaQueries
+CREATE TABLE TransformationMetaQueries(
+    TransformationID INTEGER NOT NULL,
+    MetaDataName VARCHAR(255) NOT NULL,
+    MetaDataValue BLOB NOT NULL,
+    MetaDataType VARCHAR(8) NOT NULL,
+    QueryType VARCHAR(8) NOT NULL,
```

Comment:
I think it would be easier to add the `QueryType` column to the TransformationInputDataQuery and add the `Alter table...` statement to the migration notes than to add a new table and deprecate the other one.

Reply:
Hi Andre, 
I agree with you, but In this case I would also rename the table, since 'TransformationInputDataQuery' is misleading now that also OutputQueries are included?
What do you think?